components/script/dom/filereader.rs

```diff
@@ -176,14 +191,26 @@ impl FileReader {
         fr.change_ready_state(FileReaderReadyState::Done);
         // Step 8.2
 
-        let output = match data.function {
+        let output: StringOrObject = match data.function {
```

Comment:
I have an idea that will lead to cleaner code - let's make the `perform_foo` methods take a `&self` argument, and have them store the value in `self.result`, rather than returning it.